packages/react-dom/src/__tests__/ReactDOMInput-test.js

```diff
@@ -393,40 +452,43 @@ describe('ReactDOMInput', () => {
       }
     }
 
-    const stub = ReactDOM.render(<Stub />, container);
-    const node = ReactDOM.findDOMNode(stub);
-    stub.setState({value: 0});
+    const ref = React.createRef();
+    await act(() => {
+      root.render(<Stub ref={ref} />);
+    });
+    const node = container.firstChild;
+    await act(() => {
+      ref.current.setState({value: 0});
+    });
 
     expect(node.value).toEqual('0');
   });
 
-  it('updates the value on radio buttons from "" to 0', function () {
-    ReactDOM.render(
-      <input type="radio" value="" onChange={function () {}} />,
-      container,
-    );
-    ReactDOM.render(
-      <input type="radio" value={0} onChange={function () {}} />,
-      container,
-    );
+  it('updates the value on radio buttons from "" to 0', async () => {
+    await act(() => {
+      root.render(<input type="radio" value="" onChange={function () {}} />);
+    });
+    await act(() => {
+      root.render(<input type="radio" value={0} onChange={function () {}} />);
+    });
```

Comment:
Might be worth including another test where they're in the same act, to confirm batching works. Ditto the test below